app_dart/lib/src/request_handlers/get_branches.dart

```diff
@@ -39,30 +41,43 @@ import '../service/datastore.dart';
 class GetBranches extends RequestHandler<Body> {
   GetBranches({
     required super.config,
+    required this.branchService,
     this.datastoreProvider = DatastoreService.defaultProvider,
     this.processRunner,
   });
 
+  final BranchService branchService;
   final DatastoreServiceProvider datastoreProvider;
   ProcessRunner? processRunner;
 
   static const Duration kActiveBranchActivity = Duration(days: 60);
 
+  bool isRecent(Branch b) {
+    return DateTime.now().millisecondsSinceEpoch - b.lastActivity! < kActiveBranchActivity.inMilliseconds ||
+        <String>['main', 'master'].contains(b.name);
+  }
+
   @override
   Future<Body> get() async {
     final DatastoreService datastore = datastoreProvider(config.db);
+    List<Branch> branches = await datastore.queryBranches().toList();
 
-    List<Branch> branches = await datastore
-        .queryBranches()
+    // From the dashboard point of view, these are the subset of branches we care about.
+    final RegExp branchRegex = RegExp(r'^beta|^stable|^main|^master|^flutter-.+|^fuchsia.+');
```

Comment:
Could this be simplified further? We only care about the RCs for stable and beta, which getReleaseBranches already does

Reply:
Sure, removed `beta` and `stable` from the regex. 

For context, the regex is combined with branches with recent activities to retrieve more branches beyond beta and stable.

Reply:
`getReleaseBranches` returns info for stable, beta, and latest RC which is all users care about. I'm suggesting instead of having the regex, delete it, and rely on `getReleaseBranches`

Reply:
Umm `getReleaseBranches` only return 3 branches. If we use the regex, it could retrieve branches with recent activities in the past 60 days (so that we have more than 3 options in the dropdown).

To verify my understanding, is the request here to delete the logic to fetch recent branches, and only provide 3 options from the branch selection dropdown?

Reply:
Synced with Casey, only listing 3 branches seems to be what people wanted.